spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/SleuthKafkaStreamsConfiguration.java

```diff
@@ -86,22 +87,21 @@ static KafkaStreamsBuilderFactoryBeanPostProcessor kafkaStreamsBuilderFactoryBea
 	private static final Log log = LogFactory
 			.getLog(KafkaStreamsBuilderFactoryBeanPostProcessor.class);
 
-	private final KafkaStreamsTracing kafkaStreamsTracing;
+	private final ObjectProvider<KafkaStreamsTracing> objectProvider;
 
-	KafkaStreamsBuilderFactoryBeanPostProcessor(KafkaStreamsTracing kafkaStreamsTracing) {
-		this.kafkaStreamsTracing = kafkaStreamsTracing;
+	KafkaStreamsBuilderFactoryBeanPostProcessor(ObjectProvider<KafkaStreamsTracing> objectProvider) {
+		this.objectProvider = objectProvider;
 	}
 
 	@Override
 	public Object postProcessAfterInitialization(Object bean, String beanName)
 			throws BeansException {
 		if (bean instanceof StreamsBuilderFactoryBean) {
-			StreamsBuilderFactoryBean sbfb = (StreamsBuilderFactoryBean) bean;
+			KafkaStreamsTracing kafkaStreamsTracing = this.objectProvider.getIfAvailable();
```

Comment:
this returns null, so probably deserves an if statement if no other reason clarity?